### ISSUE TITLE
Communication with dev

### ISSUE THREAD

**Issue by erplsf** [angarg12/incremental_table_elements#138]
Is there any channel to ask for information or any other conceivable way to beg for beta updates?
Joking of course, but I think your game has great potential and it would be nice to have a chat of some sort to share knowledge.
Maybe I'll write wiki page on a progression as I stumble through it.

**Comment by angarg12** [comment]
Hello!

Thank you for reaching out. Right now Github issues is the only official channel for bug reports and feature requests. For more private feedback, requests or communication you can use my gmail address (angarg12).

When the game gets release officially as v2 I may set up an official discord channel or subreddit, but at the moment I don't dare to open those just to have them deserted.

On another topic I may release a new beta version soon.

**Comment by Stevel-Knievel** [comment]
So, I figured it'd be better to leave an ask here rather than clog up the issues list, but I have a question about early game unlocks. I'm currently on day 3 of playing and I still haven't got tritium yet. Granted, I'm doing quick runs to 1e8 1H and converting to 10xH, then 2 DM. What exactly are the unlock conditions for Tritium? And thank you for making such an awesome game!

**Comment by angarg12** [comment]
You are right, probably I should set up a better/more official way of communicating. Usually I drop by the [r/incremental_games](https://discord.gg/pC9RY5B) discord channel so that's a way for the time being.

Right now, isotope unlock is proportional to your production, strictly. That means that if an isotope has abundance of 1% and your production is 5000/s, you will produce 50 isotopes per second.

Tritium is a trace element so its abundance is 0.0001% (BTW that is the lowest abundance possible for isotopes). So essentially you need a production of 1,000,000/s just to start producing tritium.

For the record, I want to change the system so that you can produce some isotopes at low probabilities, just to unlock them earlier; nevertheless you will need a lot of production to produce and then decay tritium.

How does balance and pace feels like right now? My original plan was to get to decay tritium in the first few days of playing which obviously is not happening. After the last update dramatically reduced exotic matter production I think we need to rebalance the cost of exotic upgrades to make the early game progress smoother.

**Comment by Stevel-Knievel** [comment]
The pace feels really slow in my opinion. Granted, I jumped directly from ITE to here, so the n! multiplier system to a straight 2^n system changes the flow drastically. On top of that, my main game before starting this one was realm grinder, so I'm used to quick R's after the initial grind. 

Honestly, in my opinion at least, the optimum strategy with the nerfed EM is to go straight to dark matter. I'm sitting at a 15% DM boost with Helium unlocked. What I've been doing is doing runs that last about an hour and a half or so to get to 1e7 1H, which converts to 10 xH, then to 2∅. 

My longest run that I sat on got me to 1e10 1H if I remember correctly, I remember that I had 120ish xH by the time it was done, but I used that xH on exotic upgrades I thought were permanent. 

In my limited knowledge of coding (I've been stuck solely in C++ and MATLAB), I think that the two biggest balancing changes that can be made are making isotope production a non-strict spawnrate, and by making the EM and DM generation rates piecewise functions. 

The first is to make the earlygame a bit less of a slog, and while it does depend somewhat on RNGesus, it could most likely be limited to a certain number of standard deviations from the average to prevent spawns too early/late in game. 

The second is to accelerate the growth rate of earlygame while hopefully not changing the lategame. I assume that by the mid- and lategame stages, the EM and DM amounts that an earlygame player generates would be tiny compared to the average lategame gain. Thus, it would probably be best to have an exponential/power/quadratic (whichever works best) for the 0<x<=y, then continuing with the Logistic equation from y<x<1e308, with y being whatever amount you want the changeover point to be at. That way, earlygame grows a bit faster, so people can get into the meat of the game more quickly without changing lategame balance. 

Finally, in reference to the communication front, I think a dedicated discord could be the way to go. That way there can be a link directly on the game page to direct players to a repository of information in the form of other players, and also streamline the communication and beta testing process with more people giving direct feedback. 

These are just my thoughts, I'll gladly beta test if you'd like me to as the game progresses.

**Comment by Stevel-Knievel** [comment]
Another possibility that ran through my mind (for some reason while I study for a midterm), is to have at least exotic matter based on the cumulative amount of an element generated in a prestige. That way, it feels like far less of a penalty to buy generators when the game starts to drag anchor, and it allows for faster exotic gain without changing the existing formula. 

Also, just a cosmetic thing, but I'd love to see the cyborg skin from ITE implemented here, it's nice when I'm trying to play in the middle of the night.

**Comment by angarg12** [comment]
Lots of useful feedback, thank you very much.

I'll write here a few design ramblings to share my current and future vision of pacing and balance.

At the moment I am quite pleased with the pacing of generators and upgrades, but further work needs to be done in EM and DM. Particularly, exotic upgrades need quite a big overhaul since exotic production was nerfed.

As it stands now a big chunk of early game progression comes from exotic upgrades. That is once you start unlocking x3 and x4 upgrades thing become faster. However the current costs are adjusted for the old production so obviously is going to take much longer to unlock, maybe prohibitively so.

Right now DM unlocks as soon as exotic matter and I can see how that becomes a dominant strategy. I want to change that. Ideally EM and DM would work a bit like abdications and reincarnations in Realm Grinder. I would like DM to be a bit more midgame, maybe unlocked after 1~ week of gameplay. On the other hand I am pleased with the log progression for DM. As you mention in mid-late game EM would become less of a premium currency, but I would like DM to stay a big deal for the whole game, so that almost each point counts. My current plan is to gate DM until later in game, maybe until you accumulate enough EM.

I understand your points about EM and those should be addressed next. Not only readjusting exotic upgrades costs, but production. The idea to use totals instead of current resources may sound good in early game, but it would have deep strategic implications in mid game. One of the key features of EM production is that it is the sum of all the different resources you have (for one element). What that means is that the optimal strategy is to diversify your resources as much as possible.

Only with isotopes that doesn't make sense, but when you include ions and molecules it makes a whole lot more sense. In a nutshell, when you have limited resources like ions or molecules, you have to plan how to best allocate them to maximise EM output. But if you just count the max production (instead of current value) then this doesn't become an issue anymore; you can freely cycle your resources around at no cost.

In any case I agree that early game needs to be snappier to keep people hooked. It is unacceptable to keep people idling around with nothing to do for a long while. One possibility is to lower the EM barrier one order of magnitude, so that you have more options available in early game. Another thing definitely in the TODO list is to produce isotopes earlier, even if just to tease people about more content. 

Lastly, I would like to rework the mechanics as a research three and make it a bigger part of early-to-mid game. Introduce players to isotopes and molecules earlier on so that they can play with the earlier on. This will have the added benefit that by diversifying resources will increase EM production, which in turn will speed up game progress.

Lastly, I agree that DM resetting EM is a huge setback. But instead of removing this, I would rather build the prestige system in such a way that it is less unforgiving. To be fair, if you didn't unlock decay yet, you are at a disadvantage. Once you unlock subatomic particles and energy, these ones carry on between dark prestiges. The benefit is that you are *not really* starting from scratch so progress is faster. Nevertheless I want to explore ways to make it more friendly to do a dark prestige. Probably the best way would be to have a selection of dark upgrades that speed up progress.

As you can see there is a lot to discuss in there. Right now my focus is to work on the balance and pacing of the game to make it right. I love to get feedback like yours to check whether I'm going in the right direction.

I agree about the discord channel, I will look into that.

**Comment by Stevel-Knievel** [comment]
I totally see where you're coming from in the mid- and lategame standpoint. And also, it's really refreshing to have a dev who actually tries to balance the game by implementing innovative solutions, rather than knee-jerk fixes. I wish Divine Games would take a page from your book for designing Realm Grinder. 

In terms of RG, if the EM and DM systems are supposed to resemble Abidication and Reincarnation, I would think that EM should have a higher power boost (akin to gems), and DM should give a more passive set of perks along with being used to purchase elements and such. Then, like R-power, DM could give a growing list of perks to other facets of gameplay the more one accumulates. 

But I digress. I think that the way the game is structured, having DM prestige wipe EM prestige makes perfect sense. However, as you know, with EM production being so slow, it makes this point in game feel far longer than it actually is. 

The biggest thing I thought of after reading your thoughts is the idea of a trophies tab. The entire game doesn't need to be lined out, but a visual representation of progress in the game is really nice to have to gauge strategies and such. While you do have the achievements screen, I think it would be nice to incorporate elements of the trophies and research displays from RG. I would tease new mechanics and such in the achievements tab with names, and when one is on the path to that thing, there's a representation of how close one is to that thing. Also, it's nice to have a x of y indicator on achievements with multiple levels, like the "gather hydrogen" achievement.

These are just a few of my thoughts, it's almost 4am here, so I'm sorry if this rambles a bit, but now I have a better idea of how to do stuff ingame, it's starting to rock and roll a bit more.

**Comment by angarg12** [comment]
It's funny that you mention trophies. Are you familiar with [Idle Evolution](https://www.newgrounds.com/portal/view/647724)? Early game reminds me of NG. I played it for inspiration, and one thing that I love is the 'missions' screen at the bottom. It give you very small, short-mid term goals that you can follow.

I was considering implementing something similar, only that the missions would be pooled from the achievements sections. This would have several benefits. It would give players some short and mid term goals to follow. It would encourage to achieve specific goals. It would give a sense of direction. Lastly, if you attach some rewards to these missions, you could introduce early some of the more rare resources to players.

In any case this is just a brainstorm idea and not at the top of my priority list.

**Comment by Stevel-Knievel** [comment]
Definitely, that seems like a great idea. You know where you want the game to go and you're taking it that way. I wish I could help with coding, but sadly I'm completely illiterate in js and HTML.

**Comment by angarg12** [comment]
Oh by the way, we got a discord channel and subreddit.

https://www.reddit.com/r/nucleogenesis/
https://discord.gg/wBFM7Cw

Feel free to move the discussion there. I will close this issue accordingly in a couple of days.
